AcadeMeData/forms.py

```diff
@@ -0,0 +1,22 @@
+from django import forms
+from django.contrib.auth.forms import UserCreationForm
+from AcadeMeData.models import User, DjangoUser, Degree, University
+
+
+class UserRegistrationForm(UserCreationForm):
+    university = forms.ModelChoiceField(University.objects, required=False)
+    degree = forms.ModelChoiceField(Degree.objects, required=False)
+    email = forms.EmailField(required=True)
+
+    class Meta1:
+        model = DjangoUser
+        fields = ['username', 'email', 'password1', 'password2']
+
+    def save(self, commit=True):
+        user = super(UserRegistrationForm, self).save(commit=False)
+        user.email = self.cleaned_data['email']
+        if commit:
+            user.save()
+        user2 = User(user=user, university=self.cleaned_data['university'], degree=self.cleaned_data['degree'])
```

Comment:
I don't get what is done here or why we save 2 different users

Reply:
Our 'User' model has a field that contains a 'DjangoUser', so we first create the DjangoUser and use it as input in the constructor of our user.